RT/Renderer/Backend/DX12/src/ImageReadWrite.cpp

```diff
@@ -64,4 +64,110 @@ void RT_WritePNGToDisk(const char *path, int w, int h, int channel_count, const
 		int result = stbi_write_png(path, w, h, channel_count, pixels, stride_in_bytes);
 		RT_ASSERT(result); // TODO: Cool error handling / return an error code at least
 	}
+}
+
+//  Code modified from DirectXTK12 : LoaderHelpers::LoadTextureDataFromFile (https://github.com/microsoft/DirectXTK12/blob/main/Src/LoaderHelpers.h)
+bool RT_LoadDDSImageFromDisk(RT_Arena* arena, const char* path, const uint8_t** ddsData, const struct DDS_HEADER** header, const uint8_t** bitData, size_t* bitSize)
+{
+	RT_ArenaMarker marker = RT_ArenaGetMarker(&g_thread_arena);
+
+	const uint32_t DDS_MAGIC = 0x20534444; // "DDS "
+
+	FILE* f;
+
+	if (0 == fopen_s(&f, path, "rb"))
+	{
+		long fileSize = 0;
+		
+		fseek(f, 0, SEEK_END); // seek to end of file
+		fileSize = ftell(f); // get current file pointer
+		fseek(f, 0, SEEK_SET); // seek back to beginning of file
+
+		// success opening file
+		//printf("Found and Opened DDS Texture: %s, File Size: %d\n", path, fileSize);
+
+		// check for file too big > 4294967296 (32bit) and handle
+		// Is this check necessary as fileSize is only a long?.. what happens if the file is bigger?
+		if (fileSize > 4294967296)
+		{
+			printf("Error: file too large\n");
+			return false;
+		}
+
+		// check for file too small // Need at least enough data to fill the header and magic number to be a valid DDS
+		if (fileSize < (sizeof(uint32_t) + sizeof(DDS_HEADER)))
+		{
+			printf("Error: file too small\n");
+			return false;
+		}
+
+		// read the contents of the file
+		*ddsData = (uint8_t*) malloc(fileSize);  // allocate the memory to read the file
```

Comment:
for our memory allocations we use the RT_Arena. A change will be made to replace malloc with RT_ArenaAlloc